### PR TITLE
#2176: add ACS style reaction layout

### PR DESCRIPTION
**Pull request by AleksandrParamonoff**

### CODE REVIEW

File: core/indigo-core/layout/metalayout.h [hunk 1]
```diff
@@ -21,7 +21,9 @@
 
 #include "base_cpp/obj_array.h"
 #include "base_cpp/reusable_obj_array.h"
+
 #include "math/algebra.h"
+#include <stdint.h>
```
Comment (AliaksandrDziarkach) [inline core/indigo-core/layout/metalayout.h:26]: It should be <cstdint>

File: api/c/indigo/src/option_manager.h [hunk 2]
```diff
@@ -28,6 +28,7 @@
 #include "base_cpp/red_black.h"
 
 #include <sstream>
+#include <stdint.h>
```
Comment (AliaksandrDziarkach) [inline api/c/indigo/src/option_manager.h:31]: It should be \<cstdint>

File: core/indigo-core/layout/metalayout.h (hunk 1 above)
Comment (AleksandrParamonoff) [reply core/indigo-core/layout/metalayout.h]: done

File: api/c/indigo/src/option_manager.h (hunk 2 above)
Comment (AleksandrParamonoff) [reply api/c/indigo/src/option_manager.h]: done

File: api/c/indigo/src/indigo_layout.cpp [hunk 3]
```diff
@@ -105,11 +105,9 @@ CEXPORT int indigoLayout(int object)
                              obj.type == IndigoObject::PATHWAY_REACTION || rxn.multitaleCount();
             if (!no_layout)
             {
-                ReactionLayout rl(rxn, self.smart_layout);
+                ReactionLayout rl(rxn, self.smart_layout, self.layout_options);
                 rl.max_iterations = self.layout_max_iterations;
                 rl.layout_orientation = (layout_orientation_value)self.layout_orientation;
-                rl.bond_length = MoleculeLayout::DEFAULT_BOND_LENGTH;
-                rl.horizontal_interval_factor = self.layout_horintervalfactor;
```
Comment (AliaksandrDziarkach) [inline api/c/indigo/src/indigo_layout.cpp:112]: why deleted?

File: api/c/indigo/src/indigo_options.cpp [hunk 4]
```diff
@@ -261,6 +261,62 @@ void indigoProductEnumeratorGetOneTubeMode(Array<char>& value)
...
+    static auto func = [&result](const char* mode) {
+        if (isEqual(mode, "pt"))
+        {
+            result = UnitsOfMeasure::TYPE::PT;
+        }
+        else if (isEqual(mode, "px"))
+        {
+            result = UnitsOfMeasure::TYPE::PX;
+        }
+        else if (isEqual(mode, "inches"))
```
Comment (AliaksandrDziarkach) [inline api/c/indigo/src/indigo_options.cpp:280]: should be "inch"

File: api/c/indigo/src/indigo_options.cpp [hunk 5]
```diff
@@ -261,6 +261,62 @@ void indigoProductEnumeratorGetOneTubeMode(Array<char>& value)
...
+        switch (input)
+        {
+        case UnitsOfMeasure::TYPE::PT:
+            result.readString("pt", true);
+            break;
+        case UnitsOfMeasure::TYPE::PX:
+            result.readString("px", true);
+            break;
+        case UnitsOfMeasure::TYPE::INCHES:
+            result.readString("inches", true);
```
Comment (AliaksandrDziarkach) [inline api/c/indigo/src/indigo_options.cpp:309]: should be "inch"

File: core/indigo-core/layout/metalayout.h [hunk 6]
```diff
@@ -129,6 +135,120 @@ namespace indigo
...
+    {
+        enum TYPE
+        {
+            PT,
+            PX,
+            INCHES,
+            CM
+        };
+
+        static constexpr float INCH_TO_CM = 2.54f;
```
Comment (AliaksandrDziarkach) [inline core/indigo-core/layout/metalayout.h:148]: should be CM_TO_INCH

**Review by AliaksandrDziarkach — APPROVED** [review]

**Pull request merged**